### PR TITLE
feat(reports): implement report export frontend

### PR DESCRIPTION
**Pull request by noice-noise**
## Description

- Implemented Report export frontend

## Checklist

_First, create a draft pull request. Then mark the PR as ready for review when all necessary checkbox items has been completed_

- [x] Reviewed own code
- [x] Commented on code that is hard to understand
- [ ] Implemented tests for the feature/bugfix
- [x] All **_Required_** GitHub status checks pass
- [x] The frontend and backend PR previews have been deployed

**Backend only**

- [ ] Added test data for new entity
- [ ] Generated the client api if there are new or deleted endpoints and/or DTOs

### CODE REVIEW

File: frontend/src/modules/reports/hooks/useReports.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { ReportGetDTO } from 'generated-api';
+import { useContext } from 'react';
+import { useMutation } from 'react-query';
+import { ApiContext } from '../../../shared/providers/ApiProvider';
+
+export const useReports = () => {
+  const api = useContext(ApiContext);
+
+  const getReport = useMutation((reportGetDTO: ReportGetDTO) =>
+    api.getReport(reportGetDTO, { responseType: 'blob' })
```
Comment (princhcanal) [inline frontend/src/modules/reports/hooks/useReports.ts:10]: nice 👍

File: frontend/src/shared/components/form/FormikResetButton.tsx [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { useFormikContext } from 'formik';
 import { useState, useEffect } from 'react';
 import { valuesAreEqual } from '../../helpers';
 
+/***
+ * A custom reset button that dynamically enable/disables itself
+ * during changes of form values within a Formik context
+ */
```
Comment (princhcanal) [inline frontend/src/shared/components/form/FormikResetButton.tsx:9]: ❤️

File: frontend/src/shared/helpers/clear-empty-objects.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Recursively delete any empty object members
+ * @param o target object
+ */
+export const clearEmptyObjects = (o: any) => {
```
Comment (princhcanal) [inline frontend/src/shared/helpers/clear-empty-objects.ts:5]: nice 👍 what did you use this for? seems like it's not used in the pr 🤔

**Review by princhcanal — COMMENTED** [review]
looks great! before i approve this issue though, could you have a look at the pr deployment and see why there are always errors with the notifications?

File: frontend/src/shared/helpers/clear-empty-objects.ts (hunk 3 above)
Comment (noice-noise) [reply frontend/src/shared/helpers/clear-empty-objects.ts:5]: it's been used inside the onSubmit function, it's been used to clean empty filters in the `ReportGetDTO` payload before requesting a `POST` request for the report export endpoint 😁

image

**Review by princhcanal — APPROVED** [review]
🚀

**Pull request merged**